helper/impls/helper-impl-14/src/main/java/com/linkedin/avroutil1/compatibility/avro14/codec/ParsingDecoder.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.avroutil1.compatibility.avro14.codec;
+
+import com.linkedin.avroutil1.compatibility.avro14.parsing.Parser;
+import com.linkedin.avroutil1.compatibility.avro14.parsing.SkipParser;
+import com.linkedin.avroutil1.compatibility.avro14.parsing.Symbol;
+import org.apache.avro.io.Decoder;
+
+import java.io.IOException;
+
+/** Base class for <a href="parsing/package-summary.html">parser</a>-based
+ * {@link Decoder}s. */
+public abstract class ParsingDecoder extends Decoder
+  implements Parser.ActionHandler, SkipParser.SkipHandler {
+  protected final SkipParser parser;
+
+  public ParsingDecoder(Symbol root) throws IOException {
+    this.parser = new SkipParser(root, this, this);
+  }
+
+  protected abstract void skipFixed() throws IOException;
+  
+  public void skipAction() throws IOException {
+    parser.popSymbol();
+  }
+
+  public void skipTopSymbol() throws IOException {
+    Symbol top = parser.topSymbol();
```

Comment:
Having a map like Map<Symbol, Runnable> might be faster and more readable than giant if statement.

Reply:
this code was copied out of avro. since im not 100% certain i have everything eorking i'd rather not modify it beyond the minimum needed